src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -768,7 +778,7 @@ private EncodedKey AddToPage(EncodedKey key, long value)
                 if (len == valueEncoder.Length)
                 {
                     Debug.Assert(valueEncoder.Length <= sizeof(long));
-                    Memory.Copy(b, valueEncoder.Buffer, valueEncoder.Length);
+                    Unsafe.CopyBlockUnaligned(b, valueEncoder.Buffer, (uint)valueEncoder.Length);
```

Comment:
Is that better?

Reply:
We need to eventually revisit memory copy. For small copies there is nothing faster than `.CopyBlockUnaligned()` and most likely for larger too. It has been heavily optimized with SIMD instructions and pretty fine microtuning.